src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -22,112 +16,122 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
         {
             var builder = ArrayBuilder<BoundExpression>.GetInstance();
             var stringType = GetSpecialType(SpecialType.System_String, diagnostics, node);
-            var objectType = GetSpecialType(SpecialType.System_Object, diagnostics, node);
-            var intType = GetSpecialType(SpecialType.System_Int32, diagnostics, node);
             ConstantValue? resultConstant = null;
             bool isResultConstant = true;
```

Comment:
>bool isResultConstant = true; [](start = 12, length = 29)

It looks like `isResultConstant` can be moved inside `else`.

Reply:
I'm asserting the result at the bottom, after the else. I'd prefer to keep it where it is to ensure that the assert applies to all paths through the function.

---
In reply to: [533804280](https://github.com/dotnet/roslyn/pull/49707#discussion_r533804280) [](ancestors = 533804280)

Reply:
It feels like the assert is only needed in the `else` case.

---
In reply to: [533804696](https://github.com/dotnet/roslyn/pull/49707#discussion_r533804696) [](ancestors = 533804696,533804280)